### PR TITLE
Bring back stats reporting.

### PR DESCRIPTION
**Pull request by toddboom**
Not finished yet - this brings back the anonymous reporting feature from the v0.8.x series.

### CODE REVIEW

**Comment by pauldix** [comment]
I see this is using the client. Maybe we should be writing to the old 0.8 instance we have running right now? At least until 0.9 settles a bit and we get that server upgraded.

**Comment by toddboom** [comment]
Yeah, I was wondering about that. You think we should just do a POST to the old server for now?

**Comment by pauldix** [comment]
Yeah, probably the easiest thing. We can change the reporter to post to a
new server later

**Comment by otoolep** [comment]
Don't you also need to check that the user has not disabled this? You may have to add those flags back to the config.

**Comment by toddboom** [comment]
Yeah, I still need to add that back in. I'll probably end up doing that part last.

**Comment by toddboom** [comment]
Ignore this for the moment.

**Comment by toddboom** [comment]
@pauldix @otoolep How's this look to you guys now?

File: server.go [hunk 1]
```diff
@@ -2915,3 +2916,26 @@ func copyURL(u *url.URL) *url.URL {
...
+
+func (s *Server) reportStats(version string) {
+	json := fmt.Sprintf(`[{"name":"reports", "columns":["os", "arch", "id", "version"], points:[["%s", "%s", "%x", "%s"]]}]`,
+		runtime.GOOS, runtime.GOARCH, s.ID(), version)
+
+	data := bytes.NewBufferString(json)
+
+	log.Printf("Reporting data: %s", json)
+	client := http.Client{Timeout: time.Duration(5 * time.Second)}
+	go client.Post("http://localhost:8086/db/reporting/series?u=reporter&p=influxdb", "application/json", data)
```
Comment (otoolep) [inline server.go]: I presume this URL will change.

File: server.go [hunk 2]
```diff
@@ -2915,3 +2916,26 @@ func copyURL(u *url.URL) *url.URL {
...
+	}
+}
+
+func (s *Server) reportStats(version string) {
+	json := fmt.Sprintf(`[{"name":"reports", "columns":["os", "arch", "id", "version"], points:[["%s", "%s", "%x", "%s"]]}]`,
+		runtime.GOOS, runtime.GOARCH, s.ID(), version)
+
+	data := bytes.NewBufferString(json)
+
+	log.Printf("Reporting data: %s", json)
```
Comment (otoolep) [inline server.go]: You might like to make this a little richer, so end-users don't go "what the....?". Something like "Reporting anonymized usage metrics", or something.

**Comment by otoolep** [comment]
Seems sensible. Obviously this only cover the data node (server) case.

File: server.go (hunk 1 above)
Comment (toddboom) [inline server.go]: Correct.

File: server.go (hunk 2 above)
Comment (toddboom) [inline server.go]: Good call, I'll update it.

File: server.go [hunk 3]
```diff
@@ -3311,3 +3312,49 @@ func copyURL(u *url.URL) *url.URL {
...
+  }]`, runtime.GOOS, runtime.GOARCH, version, s.ID(), clusterID, s.numSeries(), s.numMeasurements(), len(s.databases))
+
+	data := bytes.NewBufferString(json)
+
+	log.Printf("Sending anonymous usage statistics to m.influxdb.com")
+	client := http.Client{Timeout: time.Duration(5 * time.Second)}
+	go client.Post("http://m.influxdb.com:8086/db/reporting-test/series?u=reporter&p=influxdb", "application/json", data)
+}
+
+func (s *Server) numSeries() int {
```
Comment (otoolep) [inline server.go]: I believe this function, and the next, needs to RLock the server object.

Comment (otoolep) [inline server.go]: I.e.
```
s.mu.RLock()
defer s.mu.RUnlock()
```

Comment (pauldix) [inline server.go]: better to just rlock it in the `reportStats` method

Comment (otoolep) [inline server.go]: Yeah, that would be neater.

Comment (otoolep) [inline server.go]: In that case I would just move the code out of those functions, remove the functions, and just run the calculations in `reportStats`. `server.go` is getting big enough as it is. This is just a style suggestion of course.

**Comment by toddboom** [comment]
@otoolep @pauldix Ok, cleaned up based on your feedback and pointed at the live database - ready to merge on green?

**Comment by otoolep** [comment]
Looks good.

**Pull request merged**